pyfv3/_config.py

```diff
@@ -286,107 +293,59 @@ def __post_init__(self):
-            fill=namelist.fill,
-            hord_dp=namelist.hord_dp,
-            hord_mt=namelist.hord_mt,
-            hord_tm=namelist.hord_tm,
-            hord_tr=namelist.hord_tr,
-            hord_vt=namelist.hord_vt,
-            hydrostatic=namelist.hydrostatic,
-            k_split=namelist.k_split,
-            ke_bg=namelist.ke_bg,
-            kord_mt=namelist.kord_mt,
-            kord_tm=namelist.kord_tm,
-            kord_tr=namelist.kord_tr,
-            kord_wz=namelist.kord_wz,
-            n_split=namelist.n_split,
-            nord=namelist.nord,
-            npx=namelist.npx,
-            npy=namelist.npy,
-            npz=namelist.npz,
-            ntiles=namelist.ntiles,
-            nwat=namelist.nwat,
-            p_fac=namelist.p_fac,
-            rf_cutoff=namelist.rf_cutoff,
-            tau=namelist.tau,
-            vtdm4=namelist.vtdm4,
-            z_tracer=namelist.z_tracer,
-            do_qa=namelist.do_qa,
-            layout=namelist.layout,
-            grid_type=namelist.grid_type,
-            u_max=namelist.u_max,
-            do_f3d=namelist.do_f3d,
-            inline_q=namelist.inline_q,
-            do_skeb=namelist.do_skeb,
-            check_negative=namelist.check_negative,
-            tau_r2g=namelist.tau_r2g,
-            tau_smlt=namelist.tau_smlt,
-            tau_g2r=namelist.tau_g2r,
-            tau_imlt=namelist.tau_imlt,
-            tau_i2s=namelist.tau_i2s,
-            tau_l2r=namelist.tau_l2r,
-            tau_g2v=namelist.tau_g2v,
-            tau_v2g=namelist.tau_v2g,
-            sat_adj0=namelist.sat_adj0,
-            ql_gen=namelist.ql_gen,
-            ql_mlt=namelist.ql_mlt,
-            qs_mlt=namelist.qs_mlt,
-            ql0_max=namelist.ql0_max,
-            t_sub=namelist.t_sub,
-            qi_gen=namelist.qi_gen,
-            qi_lim=namelist.qi_lim,
-            qi0_max=namelist.qi0_max,
-            rad_snow=namelist.rad_snow,
-            rad_rain=namelist.rad_rain,
-            rad_graupel=namelist.rad_graupel,
-            tintqs=namelist.tintqs,
-            dw_ocean=namelist.dw_ocean,
-            dw_land=namelist.dw_land,
-            icloud_f=namelist.icloud_f,
-            cld_min=namelist.cld_min,
-            tau_l2v=namelist.tau_l2v,
-            tau_v2l=namelist.tau_v2l,
-            c2l_ord=namelist.c2l_ord,
-            regional=namelist.regional,
-            m_split=namelist.m_split,
-            convert_ke=namelist.convert_ke,
-            breed_vortex_inline=namelist.breed_vortex_inline,
-            use_old_omega=namelist.use_old_omega,
-            rf_fast=namelist.rf_fast,
-            adiabatic=namelist.adiabatic,
-            nf_omega=namelist.nf_omega,
-            fv_sg_adj=namelist.fv_sg_adj,
-            n_sponge=namelist.n_sponge,
+    def from_dict(
+        cls,
+        data: dict,
+    ) -> "DynamicalCoreConfig":
+        """Uses the nml to create a DynamicalCoreConfig, using the data
+        dictionary.
```

Comment:
There's no nml anymore at this stage. Maybe something like
```suggestion
        """Create a DynamicalCoreConfig from the given data.
```
is more appropriate?